### PR TITLE
discovery: add logging at DEBUG level for orchestrator pool predicate function when using default on-chain selection

### PR DESCRIPTION
**Pull request by kyriediculous**
**What does this pull request do? Explain your changes. (required)**
This PR adds logging at the `DEBUG` level when either `sender.ValidateTicketParams` or the broadcaster `maxPrice` check fails in the `OrchestratorPool` _predicate_ function that is used in default on-chain selection mode to filter out orchestrators that aren't usable for our B. 

This adds a little more insight when no orchestrators are found due to them all failing this _predicate_ function, currently it would only log `0 responses , err=<nil>`. 

Not necessary to include this but since I added it for debugging purposes I might as well open a PR for it. 

**How did you test each of these updates (required)**
used them to debug labrador deployment

**Does this pull request close any open issues?**

**Checklist:**
- [ ] README and other documentation updated
- [x] Node runs in OSX and devenv
- [x] All tests in `./test.sh` pass

### CODE REVIEW

File: discovery/db_discovery.go [hunk 1]
```diff
@@ -99,13 +99,25 @@ func (dbo *DBOrchestratorPoolCache) GetOrchestrators(numOrchestrators int) ([]*n
 	pred := func(info *net.OrchestratorInfo) bool {
 
 		if err := dbo.ticketParamsValidator.ValidateTicketParams(pmTicketParams(info.TicketParams)); err != nil {
+			glog.V(common.DEBUG).Infof("unable to validate ticket params - orchestrator=%v recipient=%v err=%v",
```
Comment (yondonfu) [inline discovery/db_discovery.go]: "invalid ticket params - ..." might be clearer/more concise here.

Comment (yondonfu) [inline discovery/db_discovery.go]: We use `orch=` elsewhere so let's use that here as well for consistency.

File: discovery/db_discovery.go [hunk 2]
```diff
@@ -99,13 +99,25 @@ func (dbo *DBOrchestratorPoolCache) GetOrchestrators(numOrchestrators int) ([]*n
...
 		}
 
 		// check if O's price is below B's max price
-		price := server.BroadcastCfg.MaxPrice()
-		if price != nil {
-			return big.NewRat(info.PriceInfo.PricePerUnit, info.PriceInfo.PixelsPerUnit).Cmp(price) <= 0
+		maxPrice := server.BroadcastCfg.MaxPrice()
+		price := big.NewRat(info.PriceInfo.PricePerUnit, info.PriceInfo.PixelsPerUnit)
+		if maxPrice != nil && price.Cmp(maxPrice) > 0 {
+			glog.V(common.DEBUG).Infof("orchestrator's price is too high - orchestrator=%v recipient=%v price=%v maxPrice=%v",
```
Comment (yondonfu) [inline discovery/db_discovery.go]: We use `orch=` elsewhere so let's use that here as well for consistency.

File: discovery/db_discovery.go [hunk 3]
```diff
@@ -99,13 +99,25 @@ func (dbo *DBOrchestratorPoolCache) GetOrchestrators(numOrchestrators int) ([]*n
...
+				info.GetTranscoder(),
+				ethcommon.BytesToAddress(info.GetTicketParams().GetRecipient()),
+				price.FloatString(3),
```
Comment (yondonfu) [inline discovery/db_discovery.go:115]: wei/pixel suffix for the prices?

File: discovery/db_discovery.go [hunk 4]
```diff
@@ -99,13 +99,25 @@ func (dbo *DBOrchestratorPoolCache) GetOrchestrators(numOrchestrators int) ([]*n
...
 		// check if O's price is below B's max price
-		price := server.BroadcastCfg.MaxPrice()
-		if price != nil {
-			return big.NewRat(info.PriceInfo.PricePerUnit, info.PriceInfo.PixelsPerUnit).Cmp(price) <= 0
+		maxPrice := server.BroadcastCfg.MaxPrice()
+		price := big.NewRat(info.PriceInfo.PricePerUnit, info.PriceInfo.PixelsPerUnit)
+		if maxPrice != nil && price.Cmp(maxPrice) > 0 {
+			glog.V(common.DEBUG).Infof("orchestrator's price is too high - orchestrator=%v recipient=%v price=%v maxPrice=%v",
+				info.GetTranscoder(),
+				ethcommon.BytesToAddress(info.GetTicketParams().GetRecipient()),
```
Comment (yondonfu) [inline discovery/db_discovery.go]: Need a `.Hex()` or else this will just print a slice

File: discovery/db_discovery.go (hunk 1 above)
Comment (yondonfu) [inline discovery/db_discovery.go]: Not sure how useful including the recipient address is here. Also need to use `.Hex()` if we do include it or else we'll just print a slice.

**Review by yondonfu — CHANGES_REQUESTED** [review]

File: discovery/db_discovery.go (hunk 1 above)
Comment (kyriediculous) [reply discovery/db_discovery.go]: Yeah at this point I'm not certain what our go-to orchestrator identification method is: by address or by service URI. I assume service URI would be a better fit as most will eventually end up using a domain name.

**Comment by kyriediculous** [comment]
Submitted change requests and removed recipient address in favor of orchestrator service uri fc7677c7

File: discovery/db_discovery.go [hunk 5]
```diff
@@ -99,13 +99,25 @@ func (dbo *DBOrchestratorPoolCache) GetOrchestrators(numOrchestrators int) ([]*n
 	pred := func(info *net.OrchestratorInfo) bool {
 
 		if err := dbo.ticketParamsValidator.ValidateTicketParams(pmTicketParams(info.TicketParams)); err != nil {
+			glog.V(common.DEBUG).Infof("invalid ticket params - orch=%v err=%v",
+				info.GetTranscoder(),
+				ethcommon.BytesToAddress(info.GetTicketParams().GetRecipient()).Hex(),
```
Comment (yondonfu) [inline discovery/db_discovery.go]: We can remove this.

File: discovery/db_discovery.go [hunk 6]
```diff
@@ -99,13 +99,25 @@ func (dbo *DBOrchestratorPoolCache) GetOrchestrators(numOrchestrators int) ([]*n
...
 		// check if O's price is below B's max price
-		price := server.BroadcastCfg.MaxPrice()
-		if price != nil {
-			return big.NewRat(info.PriceInfo.PricePerUnit, info.PriceInfo.PixelsPerUnit).Cmp(price) <= 0
+		maxPrice := server.BroadcastCfg.MaxPrice()
+		price := big.NewRat(info.PriceInfo.PricePerUnit, info.PriceInfo.PixelsPerUnit)
+		if maxPrice != nil && price.Cmp(maxPrice) > 0 {
+			glog.V(common.DEBUG).Infof("orchestrator's price is too high - orch=%v price=%v wei/pixel maxPrice=%v wei/pixel",
+				info.GetTranscoder(),
+				ethcommon.BytesToAddress(info.GetTicketParams().GetRecipient()).Hex(),
```
Comment (yondonfu) [inline discovery/db_discovery.go]: We can remove this.

**Review by yondonfu — CHANGES_REQUESTED** [review]

File: discovery/db_discovery.go (hunk 6 above)
Comment (kyriediculous) [reply discovery/db_discovery.go]: fixed in 3829307

File: discovery/db_discovery.go (hunk 5 above)
Comment (kyriediculous) [reply discovery/db_discovery.go]: fixed in 3829307

**Review by yondonfu — COMMENTED** [review]
Changes look good. Let's rebase

**Comment by kyriediculous** [comment]
rebased !

**Review by yondonfu — APPROVED** [review]
LGTM!

**Pull request merged**